.travis.yml

```diff
@@ -1,39 +1,47 @@
 sudo: false
-cache: pip
+cache:
+  apt: true
+  directories:
+    - $HOME/.cache/pip
 
 addons:
   apt:
    sources:
       - pov-wkhtmltopdf
    packages:
       - expect-dev  # provides unbuffer utility
-      - python-lxml # because pip installation is slow
       - wkhtmltopdf
 
 language: python
 
 python:
   - "2.7"
 
-env:
-  - VERSION="9.0" LINT_CHECK="1"
-  - VERSION="9.0" ODOO_REPO="odoo/odoo" LINT_CHECK="0"
-  - VERSION="9.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0"
-
 virtualenv:
   system_site_packages: true
 
+env:
+  global:
+    - DB_NAME=${TRAVIS_JOB_ID}
+    - ADDONS_INST=cmis
+    - ADDONS_TEST=cmis_field,cmis_af,cmis_web,cmis_web_alf
+
 before_install:
   - "export DISPLAY=:99.0"
   - "sh -e /etc/init.d/xvfb start"
+  - pip install acsoo
+  - pip install -q QUnitSuite coveralls codecov
+  - pip install -r requirements.txt
+  - pip install -e setup/cmis_field
+  - pip install -e setup/cmis_alf
+  - pip install -e setup/cmis_web
+  - pip install -e setup/cmis_web_alf
```

Comment:
For the sake of creating a generic example, instead of installing all addons manually, you could do something like this:
https://github.com/acsone/maintainer-quality-tools/blob/acf57440f4dc821c03212dd398580fb8fb022da6/travis/travis_install_nightly#L93-L95